src/components/badge/index.tsx

```diff
@@ -1,109 +1,115 @@
 import _ from 'lodash';
-import PropTypes from 'prop-types';
-import React from 'react';
-import {StyleSheet, Text, ViewPropTypes} from 'react-native';
+import React, {PureComponent} from 'react';
+import {ImageSourcePropType, ImageStyle, StyleProp, StyleSheet, Text, TextStyle, TouchableOpacityProps, ViewStyle, ViewProps} from 'react-native';
 import {View as AnimatableView} from 'react-native-animatable';
-import {PureBaseComponent} from '../../commons';
+import {extractAccessibilityProps, extractAnimationProps} from '../../commons/modifiers';
+import {asBaseComponent} from '../../commons/new';
 import {BorderRadiuses, Colors, ThemeManager, Typography} from '../../style';
 import Image from '../image';
 import TouchableOpacity from '../touchableOpacity';
 import View from '../view';
 
-const LABEL_FORMATTER_VALUES = [1, 2, 3, 4];
+const LABEL_FORMATTER_VALUES = [1, 2, 3, 4] as const;
 
 // TODO: depreciate enum badge sizes, use only number for size
-export const BADGE_SIZES = {
-  pimpleSmall: 6,
-  pimpleBig: 10,
-  pimpleHuge: 14,
-  small: 16,
-  default: 20,
-  large: 24
+export enum BADGE_SIZES {
+  pimpleSmall = 6,
+  pimpleBig = 10,
+  pimpleHuge = 14,
+  small = 16,
+  default = 20,
+  large = 24
 };
 
+type LabelFormatterValues = typeof LABEL_FORMATTER_VALUES[number]
+export type BadgeSizes = keyof typeof BADGE_SIZES
+
+export type BadgePropTypes = ViewProps & TouchableOpacityProps & {
+  /**
+   * Text to show inside the badge.
+   * Not passing a label (undefined) will present a pimple badge.
+   */
+  label?: string;
+  /**
+   * Color of the badge background
+   */
+  backgroundColor?: string;
+  /**
+   * the badge size (default, small)
+   */
+  size: BadgeSizes | number;
```

Comment:
Should it be optional as the rest of the props? 
Otherwise it will require user to pass it (even though it has a default value)

Reply:
I've checked the example screen and it doesn't require since it knows to use the `defaultProp` (I left one example where size is not passed).
As far as I know this is the only way to make TS take into account `defaultProps`:
` size: 'default'`.
Also, TS doesn't like it as optional since yells at me for methods that use `size` and assume that it can be undefined and fall back to `defaultProp`. 

Same thing we did in Avatar btw: https://github.com/wix/react-native-ui-lib/blob/master/src/components/avatar/index.tsx#L121